### PR TITLE
Refactor schur_complement operator

### LINKED ISSUE #3913: [8.5.0 release] Test lac/schur_complement_05 fails
$ ctest -R "linear_operator|linear-operator|schur_complement"
...
94% tests passed, 4 tests failed out of 70
The following tests FAILED:
	319 - lac/schur_complement_05.mpirun=1.debug (Failed)
	320 - lac/schur_complement_05.mpirun=1.release (Failed)
	321 - lac/schur_complement_05.mpirun=3.debug (Failed)
	322 - lac/schur_complement_05.mpirun=3.release (Failed)
Errors while running CTest

### PR DESCRIPTION
**Pull request by jppelteret**
Subsequent to the implementation of Payloads for LinearOperators, it is
necessary that compound operations such as this one be expressed in
terms of overloaded operations (operator+,* etc.) instead of in a
low-level format, with the operator functions v_mult etc. expanded
manually. This is because the payload is not correctly configured in the
latter case, and the operations defined on the Payload-level would not
align with those being performed by the higher-level LinearOperator
interface. When operator+,* etc. are called, the equivalent operations
for the payload are automatically set up.

Fixes #3913

### CODE REVIEW

File: tests/lac/schur_complement_05.with_cxx11=on.with_trilinos=true.with_mpi=true.mpirun=1.output [hunk 1]
```diff
@@ -6,10 +6,10 @@ DEAL:ManualSchur:cg::Convergence step 11 value 0
 DEAL:SchurComplementOp:Pre::Starting value 12.7149
 DEAL:SchurComplementOp:Pre::Convergence step 43 value 1.05660e-05
 DEAL:SchurComplementOp:Main::Starting value 7.49608e-06
-DEAL:SchurComplementOp:Main::Convergence step 12 value 4.00858e-10
+DEAL:SchurComplementOp:Main::Convergence step 22 value 2.93110e-10
```
Comment (jppelteret) [inline tests/lac/schur_complement_05.with_cxx11=on.with_trilinos=true.with_mpi=true.mpirun=1.output]: This looks dodgy to me. I'm still investigating why the number of solver iterations has increased.

Comment (tamiko) [inline tests/lac/schur_complement_05.with_cxx11=on.with_trilinos=true.with_mpi=true.mpirun=1.output]: I wouldn't be too surprised by that. Going down to a residual of 10e-10 might - depending on the initial residual - be very well within round-off error.
Then, it very much depends on the compiler and the degree of optimization how many iterations you get.

Comment (tamiko) [inline tests/lac/schur_complement_05.with_cxx11=on.with_trilinos=true.with_mpi=true.mpirun=1.output]: If you want to examine this - simply print out per round reduction values. I bet that between around step 12 and 22 you don't see much convergence but simply a residual lingering slightly above 10e-9 (or something on that line).

Comment (jppelteret) [inline tests/lac/schur_complement_05.with_cxx11=on.with_trilinos=true.with_mpi=true.mpirun=1.output]: Thanks for the tip. I'll look at this a little more closely tomorrow and will report back then. I was trying my hardest to reproduce the results attained with our own solver (the benchmark computation uses `dealii::SolverCG<TrilinosWrappers::MPI::Vector>`), but maybe this is not possible. Trilinos' solvers seem to be a bit more quirky than our own implementations.

**Comment by tamiko** [comment]
@jppelteret Jup, iteration is unstable.
```
5177: ! DEAL:ManualSchur:cg::Starting value 5.38429e-06
5177:   DEAL:ManualSchur:cg::Convergence step 11 value 0
5177:   DEAL:SchurComplementOp:Pre::Starting value 16.9566
5177: ! DEAL:SchurComplementOp:Pre::Convergence step 59 value 1.46447e-05
5177: ! DEAL:SchurComplementOp:Main::Starting value 5.38429e-06
5177: ! DEAL:SchurComplementOp:Main::Convergence step 34 value 8.46507e-10
5177:   DEAL:SchurComplementOp:Post::Starting value 16.9566
5177: ! DEAL:SchurComplementOp:Post::Convergence step 105 value 1.46447e-05
5177: ! DEAL::Rel. norm of error in SchurOperator solve: 0.000266920
```

**Comment by jppelteret** [comment]
Thanks @tamiko, thats very useful to know. I suppose I'll have to refactor the test and evaluate it with some more reproducible metric.

**Comment by tamiko** [comment]
@jppelteret You already compare the relative norms of the residuals. That should be enough.

**Comment by tamiko** [comment]
Even more, the relative error, not the relative norms.

**Comment by jppelteret** [comment]
Then I'll just strip out the logging of the linear solver :-)

**Comment by tamiko** [comment]
@jppelteret I will push a patch.

**Comment by jppelteret** [comment]
As you wish. I need about 30 minutes until I can do it myself.

**Comment by tamiko** [comment]
@jppelteret Can you please cherry-pick the last commit on https://github.com/jppelteret/dealii/pull/2 :-)

**Comment by jppelteret** [comment]
Thanks a lot @tamiko, I will do! Much appreciated :-)

**Comment by jppelteret** [comment]
@tamiko Do you want me to squash all of the test updates into one commit?

**Comment by tamiko** [comment]
@jppelteret Yes please. After that please merge.

**Pull request merged**